### ISSUE TITLE
Normalization layer masking

### ISSUE THREAD

**Issue by ianzur** [tensorflow/tensorflow#52234]
**System information**
- TensorFlow version (you are using): nightly (on colab) (also on my machine, but behind a few weeks (built upstream from source for ROCm support))
- Are you willing to contribute it (Yes/No): Yes, once maintainers evaluate proposal

**Describe the feature and the current behavior/state.**
tf.keras.layers.Normalization() does not ignore masked values.

**Will this change the current api? How?**
Only additional parameters, should not contain any non-backwards compatible changes.

**Who will benefit with this feature?**
People working with ragged data who want to include a preprocessing layer to normalize in model.

**Any Other info.**

Proposing:
- add parameter to `.adapt()` method (mask_value=0.0 and/or boolean mask)
- add `call(..., mask=None)` and tensor multiply to allow masking during model fitting/evaluation

[Colab gist current behavior](https://colab.research.google.com/drive/1aKGeS0lJy-QEQx0jARyOAqAMFBEFRe--?usp=sharing)

An attempt at masked normalization (does **NOT** generalize to any input shape) 
```
class MaskedNormalization(tf.keras.layers.Layer):
    """custom masked normalization layer"""

    def __init__(
        self,
        mask_value: int = 999,
        input_dims: tuple = (51, 3),
        name: str = "masked_normalizer",
    ):
        super(MaskedNormalization, self).__init__()

        self.mask_value = mask_value
        self.supports_masking = True
 
        self.normalization = tf.keras.layers.experimental.preprocessing.Normalization(axis=2)


    def adapt(self, inputs):
        """redefined adapt to ignore masked values"""
        if not self.normalization.built:
            raise RuntimeError("layer must be \"built\" prior to calling `.adapt()`, see `model.build_graph()`")

        mask = inputs != self.mask_value

        float_mask = tf.cast(mask, dtype=tf.dtypes.float32)

        mean = tf.reduce_sum(inputs * float_mask, axis=[0, 1]) / tf.reduce_sum(
            float_mask, axis=[0, 1]
        )
        # sample variance (divide by: n-1)
        var = tf.reduce_sum(((inputs - mean) ** 2) * float_mask, axis=[0, 1]) / (
            tf.reduce_sum(float_mask, axis=[0, 1])
            - tf.constant(1, dtype=tf.dtypes.float32, shape=(3,))
        )
        count = tf.experimental.numpy.count_nonzero(
            float_mask
        )

        self.normalization.adapt_mean.assign(mean)
        self.normalization.adapt_variance.assign(var)
        self.normalization.count.assign(count)

        self.normalization.finalize_state()
        self.normalization._is_adapted = True

    def call(self, inputs, mask=None):

        # if mask provided mask output values (multiply by zero)
        if mask is not None:
                reshape_mask = tf.transpose(
                    tf.expand_dims(tf.cast(mask, dtype=tf.dtypes.float32), axis=0),
                    perm=[1, 2, 0],
                )
            return self.normalization(inputs) * reshape_mask

        return self.normalization(inputs)

    def compute_mask(self, inputs, mask=None):
        """return the input_mask directly"""
        return mask
```

**Comment by mohantym** [comment]
Hi @ianzur !
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!

**Comment by ianzur** [comment]
Moved issue to keras repo.

Closing here.
